### PR TITLE
ZK-4922: zEmbedded upload fails

### PR DESCRIPTION
**Pull request by DevChu**
ZK-4923: zEmbedded download fails

### CODE REVIEW

File: zk/src/archive/web/js/zk/au.ts [hunk 1]
```diff
@@ -1394,6 +1394,11 @@ zAu.cmd0 = /*prototype*/ { //no uuid at all
 	 */
 	download: function (url) {
 		if (url) {
+			if (window['zEmbedded']) {
+				var dt = zk.Desktop.$(),
+					updateURI = dt.updateURI;
+				url = updateURI.substr(0, updateURI.indexOf(dt.contextURI)) + url;
```
Comment (DevChu) [inline zk/src/archive/web/js/zk/au.ts:1400]: substring

Comment (DevChu) [reply zk/src/archive/web/js/zk/au.ts:1400]: use general function (look into zk host url)

**Review by scribetw — CHANGES_REQUESTED** [review]
refine needed

File: zktest/src/archive/test2/B96-ZK-4923-1.zul [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        </library-property>
+        3. add cors filter in web.xml.
+        4. Open "http://localhost:9000" to test. (container <A>)
+    ]]></label>
+	<label>The following source should work fine (load with absolute path)</label>
+	<separator />
+	<style src="/widgets/grid/data_binding/Customize.css"/>
+	<script src=""/>
+	<a href="/">To zk test</a>
+	<image src="/img/sun.jpg"/>
```
Comment (DevChu) [inline zktest/src/archive/test2/B96-ZK-4923-1.zul:31]: + ~./
+ ../

Comment (DevChu) [reply zktest/src/archive/test2/B96-ZK-4923-1.zul:31]: remove other comps

**Review by scribetw — CHANGES_REQUESTED** [review]
test case should be refined

**Review by plijyfes — APPROVED** [review]

File: zktest/src/archive/test2/B96-ZK-4923-1.zul [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <name>org.zkoss.web.servlet.http.embedded.enabled</name>
+            <value>true</value>
+        </library-property>
+        3. add cors filter in web.xml.
+        4. Open "http://localhost:9000" to test. (container <A>)
+    ]]></label>
+	<label>You should see 5 images and 1 button with image (load with absolute path)</label>
+	<separator />
+	<image src="/img/sun.jpg"/>
+	<image src="./img/corner.gif"/>
```
Comment (DevChu) [inline zktest/src/archive/test2/B96-ZK-4923-1.zul:29]: src="xx.png" (current path)

Comment (DevChu) [reply zktest/src/archive/test2/B96-ZK-4923-1.zul:29]: reverse condition, if no absolute -> need add prefix

**Review by scribetw — APPROVED** [review]

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**